packages/react/src/hooks/useStorageURL.ts

```diff
@@ -14,31 +14,35 @@ export interface UseStorageURLResult {
  */
 export const useStorageURL = (
   key: string,
-  options?: S3ProviderGetConfig
+  options?: S3ProviderGetConfig,
+  fallbackURL?: string
 ): UseStorageURLResult & { fetch: () => () => void } => {
   const [result, setResult] = React.useState<UseStorageURLResult>({
     isLoading: true,
   });
 
   // Used to prevent an infinite loop on useEffect, because `options`
   // will have a different reference on every render
-  const serializedOptions = JSON.stringify(options);
+  const optionsRef = React.useRef(options);
 
   const fetch = () => {
     setResult({ isLoading: true });
 
+    const options = optionsRef.current;
     const promise = Storage.get(key, options);
 
     // Attempt to fetch storage object url
     promise
       .then((url) => setResult({ url, isLoading: false }))
-      .catch((error: Error) => setResult({ error, isLoading: false }));
+      .catch((error: Error) =>
+        setResult({ url: fallbackURL, error, isLoading: false })
```

Comment:
Do we need to verify the validity of `fallbackUrl`?

Reply:
Think that will require a new HTTP request,. Since the `img` tag will try to make the request anyway and triggers `onError` if that's invalid, think should be sufficient.

Reply:
Can we add a unit test ensuring this behavior?

Reply:
Sure

Reply:
I don't think we need to test this behavior as it's standard browser behavior and we can't control whether customer provide a valid fallbackUrl.